### PR TITLE
Use DataView to access WebAssembly memory

### PR DESCRIPTION
**Pull request by Josh-Cena**
Accompanying https://github.com/mdn/content/pull/33960

### CODE REVIEW

File: js-api-examples/memory.html [hunk 1]
```diff
@@ -11,14 +11,14 @@
...
-      WebAssembly.instantiateStreaming(fetch("memory.wasm"), { js: { mem: memory } })
-      .then(obj => {
-        const summands = new Uint32Array(memory.buffer);
+      WebAssembly.instantiateStreaming(fetch("memory.wasm"), {
+        js: { mem: memory },
+      }).then((obj) => {
+        const summands = new DataView(memory.buffer);
         for (let i = 0; i < 10; i++) {
-          summands[i] = i;
+          summands.setUint32(i * 4, i, true);
```
Comment (hamishwillee) [inline js-api-examples/memory.html:21]:
```suggestion
          summands.setUint32(i * 4, i, true); // WebAssembly is little endian
```

Comment (hamishwillee) [reply js-api-examples/memory.html:21]: Just to match the docs

**Review by hamishwillee — APPROVED** [review]
Also approved so you can self merge. There might be a reason not to update this as I have suggested.

**Comment by Josh-Cena** [comment]
Sorry I don't have write access to this repo, so you would have to do the merging

**Comment by bsmth** [comment]
> you would have to do the merging

I'll go ahead and merge to push it a long! Thanks, both!

**Pull request merged**